research/03-literate-prog.tex

```diff
@@ -1,6 +1,109 @@
+\newpage
 \section{Literate Programming}
 \label{sec:literate-programming}
 
+Literate programming was first introduced in 1984 for the Pascal
+language~\cite{knuth1984}. Even though the idea was conceived over 30 years
+ago, implementations are not very common. In recent years the idea seems to
+gain popularity again. Examples include the IPython project~\cite{ipython2007},
+published in 2007, as well as Swift Playgrounds~\cite{swift-playgrounds},
+introduced in 2014. 
+Just as with REPLs, the concept of literate programming is known by various
+names in slightly different forms. In this section it is first explained what
+exactly literate programming entails.
+
+\subsection{Literate programming}
+According to Donald Knuth better documentation of programs is essential to make
+further progress in the state of the art of programming.
+To achieve this he proposes to write programs not with the intention to
+explain the computer what to do, but with the intention to explain to humans
+what the programmer wants a computer to do~\cite{knuth1984}.
+He proposes to do this by mixing documentation and code in a single source file.
+This idea of `literate programming' was realized in its original form as the
+`WEB' language developed during D. Knuth's research at Stanford University.
+
+\subsubsection{The `WEB' language}
+The `WEB' language was mainly a combination of two other languages,
+namely \TeX{} for document formatting and PASCAL as the programming language.
+By combining these languages, each with a different purpose, into one language,
+the source code serves two purposes:
+\begin{itemize}
+  \item it produces a document describing the program
+  \item it produces a machine-executable program
+\end{itemize}
+Since both program and documentation are generated from the same source file,
+they are always consistent with each other.
+
+As a follow up another language called `CWEB' was developed, introducing
```

Comment:
I think it's better to have this instead: `After `WEB`, another language called `CWEB` was developed to introduce the same literate programming style to the C programming language.`